dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/utils/RestfulWSUtils.java

```diff
@@ -0,0 +1,277 @@
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HEAD;
+import javax.ws.rs.OPTIONS;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.jboss.resteasy.core.ResourceMethodRegistry;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+
+import io.openliberty.restfulWS.introspector.RESTfulEndpointLoggingIntrospector;
+
+/**
+ * A collection of restfulWS-3.0 utility methods.
+ */
+public class RestfulWSUtils {
+
+    private static final TraceComponent tc = Tr.register(RestfulWSUtils.class);
+
+    /*
+     * Register the servlet's restfulWS endpoints with the ResteasyEndpointIntrospector
+     * and print them to messages.log if trace is enabled
+     */
+    @SuppressWarnings("rawtypes")
+    @FFDCIgnore(NoClassDefFoundError.class)
+    public static void logEndpoints(ServletConfig servletConfig) {
+
+        RESTfulEndpointLoggingIntrospector introspector = getRESTfulEndpointLoggingIntrospector();
+
+        if (introspector != null) {
+
+            ServletContext sc = servletConfig.getServletContext();
+            String appname = sc.getServletContextName();
+            String contextRoot = sc.getContextPath();
+            String servletName = servletConfig.getServletName();
+            String servletMapping = servletConfig.getInitParameter("resteasy.servlet.mapping.prefix");
+            String resourceList = servletConfig.getInitParameter("resteasy.scanned.resources");
+            String[] resources = null;
+            if (resourceList != null) {
+                resources = resourceList.split(",");
+            }
+            StringBuffer sb = new StringBuffer();
+            sb.append("Endpoints for Application: ").append(appname).append(" - ").append(servletName).append("\n");
+            
+            if (resources != null) {
+                for (String clazz : resources) {
+
+                    // load the class
+                    Class resource = loadClass(clazz, sb);
+                    if (resource != null) {
+
+                        sb.append(resource.getName()).append("\n");
+
+                        // check if the class is annotated with @Path, if so, get the value
+                        String classPath = getAnnotationValue(resource, Path.class, String.class);
+
+                        // get the @Consumes MediaTypes
+                        String classConsumes[] = getAnnotationValue(resource, Consumes.class, String[].class);
```

Comment:
```suggestion
                        String[] classConsumes = getAnnotationValue(resource, Consumes.class, String[].class);
```

This looks like C style arrays - Java styling prefers the bracket to be near to the type to make clear that the type is an array of Strings.